### PR TITLE
allow use of `test_cmd` without `runtest` for `ConfigureMake`

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

When using `CMakeNinja` and running tests one needs `test_cmd = 'ctest'` and `runtest = ''` which is akward but required because it simply concats `runtest` even when it is not set (`None`)

Also handle the default case for `test_cmd` in the test function so derived classes can detect if the command was set by the user or not. Required for #2838 

And finally clean up the executed command by omitting empty values (e.g. pre/posttestopts not set which resulted in `" ctest  "`)

### CODE REVIEW

File: easybuild/easyblocks/generic/configuremake.py [hunk 1]
```diff
@@ -350,13 +351,13 @@ def test_step(self):
...
 
         test_cmd = self.cfg.get('test_cmd') or DEFAULT_TEST_CMD
-        if self.cfg['runtest'] or test_cmd != DEFAULT_TEST_CMD:
-            cmd = ' '.join([
-                self.cfg['pretestopts'],
-                test_cmd,
-                self.cfg['runtest'],
-                self.cfg['testopts'],
-            ])
+        run_test = self.cfg['runtest']
```
Comment (boegel) [inline easybuild/easyblocks/generic/configuremake.py:354]: better use `runtest` to align with the easyconfig parameter name?

File: easybuild/easyblocks/generic/configuremake.py [hunk 2]
```diff
@@ -350,13 +351,13 @@ def test_step(self):
...
+        if run_test or test_cmd != DEFAULT_TEST_CMD:
+            # Make run_test a string (empty if it is e.g. a boolean)
+            if not isinstance(run_test, str):
```
Comment (boegel) [inline easybuild/easyblocks/generic/configuremake.py:357]: Don't use `str` directly, use `string_type` from `py2vs3`:

```python
from easybuild.tools.py2vs3 import string_type
...
             if not isinstance(runtest, string_type):

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/generic/configuremake.py (hunk 2 above)
Comment (Flamefire) [reply easybuild/easyblocks/generic/configuremake.py:357]: I wanted to stick to what is used at other places in the same file, e.g. line [331](https://github.com/easybuilders/easybuild-easyblocks/blob/077a9036be1b5666a50bef1b746ec7834f256900/easybuild/easyblocks/generic/configuremake.py#L331)

Shall I still replace this and maybe all others in this file too?

File: easybuild/easyblocks/generic/configuremake.py (hunk 1 above)
Comment (Flamefire) [reply easybuild/easyblocks/generic/configuremake.py:354]: Makes sense

File: easybuild/easyblocks/generic/configuremake.py (hunk 2 above)
Comment (boegel) [reply easybuild/easyblocks/generic/configuremake.py]: It's better to use `string_type` consistently, there could be surprises when using `str` w.r.t. behavior in Python 2 vs 3...

Comment (Flamefire) [reply easybuild/easyblocks/generic/configuremake.py]: Ok, done

**Comment by Flamefire** [comment]
Anything missing here?

**Comment by boegel** [comment]
> Anything missing here?

@Flamefire Similar to #2838: a battery of tests with existing easyconfigs that either use `ConfigureMake` directly, or an easyblock that derives from it, mostly.

**Comment by Flamefire** [comment]
I started test runs for both with some random ECs using ConfigureMake/CMakeMake respectively, hope it doesn't fail for unrelated reasons as I don't expect any issues by these PRs as explained in https://github.com/easybuilders/easybuild-easyblocks/pull/2838#issuecomment-1342777832

**Comment by Flamefire** [comment]
Test report by @Flamefire
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _arpack-ng-3.8.0-foss-2022b.eb_ 
 * **SUCCESS** _QwtPolar-1.1.1-GCCcore-10.2.0.eb_ 
 * **SUCCESS** _ICU-69.1-GCCcore-11.2.0.eb_ 
 * **SUCCESS** _ICU-72.1-GCCcore-12.2.0.eb_ 
 * **SUCCESS** _Vim-9.0.0950-GCCcore-11.3.0.eb_ 
 * **SUCCESS** _cURL-7.78.0-GCCcore-11.2.0.eb_ 
 * **SUCCESS** _cURL-7.83.0-GCCcore-11.3.0.eb_ 
 * **SUCCESS** _cURL-7.86.0-GCCcore-12.2.0.eb_ 
 * **SUCCESS** _cairo-1.16.0-GCCcore-11.2.0.eb_ 
 * **SUCCESS** _c-ares-1.18.1-GCCcore-11.2.0.eb_ 
 * **SUCCESS** _CoinUtils-2.11.6-GCC-11.2.0.eb_ 
 * **SUCCESS** _libdap-3.20.8-GCCcore-11.2.0.eb_ 
 * **SUCCESS** _libxml2-python-2.9.13-GCCcore-11.3.0.eb_ 
 * **SUCCESS** _xproto-7.0.31-GCCcore-11.2.0.eb_ 
 * **SUCCESS** _ITSTool-2.0.7-GCCcore-11.3.0.eb_ 
 * **SUCCESS** _imake-1.0.8-GCCcore-11.2.0.eb_ 

Build succeeded for 16 out of 16 (14 easyconfigs in total)
taurusi8034 - Linux CentOS Linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor, 8 x NVIDIA NVIDIA A100-SXM4-40GB, 470.57.02, Python 2.7.5
See https://gist.github.com/Flamefire/9c92d1fabdd65cac48c620fe6bef8771 for a full test report.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Test report by @akesandgren
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Amber-22.0-foss-2021b-AmberTools-22.3.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
b-an02.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 3.8.10
See https://gist.github.com/akesandgren/bcccb9a6f0910ea04a240647e915a94a for a full test report.

**Comment by akesandgren** [comment]
Going in, thanks @Flamefire!

**Pull request merged**